src/Sylius/Behat/Context/Api/Shop/CheckoutContext.php

```diff
@@ -398,7 +396,7 @@ public function iTryToSelectShippingMethod(string $shippingMethodCode): void
             HTTPRequest::METHOD_PATCH,
             sprintf('shipments/%s', $this->getCart()['shipments'][0]['id']),
         );
-        $request->setContent(['shippingMethod' => $this->iriConverter->getItemIriFromResourceClass($this->shippingMethodClass, ['code' => $shippingMethodCode])]);
+        $request->setContent(['shippingMethod' => sprintf('/api/v2/shop/shipping-methods/%s', $shippingMethodCode)]);
```

Comment:
```suggestion
       $request->setContent(['shippingMethod' => $this->iriConverter->getItemIriFromResourceClass($shippingMethod)]);
```
I would use transform to retrieve the shippingMethod directly and use it here